src/app/destinyTrackerApi/d2-bulkFetcher.ts

```diff
@@ -53,14 +53,27 @@ class D2BulkFetcher {
       return Promise.resolve<D2ItemFetchResponse[]>([]);
     }
 
-    const promise = dtrFetch(
-      `https://db-api.destinytracker.com/api/external/reviews/fetch?platform=${platformSelection}&mode=${mode}`,
-      itemList
-    ).then(handleD2Errors, handleD2Errors);
+    const size = 10;
+    const arrayOfArrays: D2ItemFetchRequest[][] = [];
+    for (let i = 0; i < itemList.length; i += size) {
+      arrayOfArrays.push(itemList.slice(i, i + size));
+    }
+
+    const results: D2ItemFetchResponse[] = [];
+
+    for (const arraySlice of arrayOfArrays) {
+      const promiseSlice = dtrFetch(
+        `https://db-api.destinytracker.com/api/external/reviews/fetch?platform=${platformSelection}&mode=${mode}`,
+        arraySlice
+      ).then(handleD2Errors, handleD2Errors);
 
-    loadingTracker.addPromise(promise);
+      loadingTracker.addPromise(promiseSlice);
+
+      const result = await promiseSlice;
+      results.push(...result);
+    }
 
-    return promise;
+    return results;
```

Comment:
I put a try/catch block around the response `await` in the loop.

I don't know how we generally guard against failures on optional resource fetches.

Reply:
My point was more that the success rate of fetching ratings is now `(number of items / 10) * (API failure rate)`. It might be good to handle individual batch failures and retry them, for example, or return what you've gotten so far so you don't need to reload them next time.

Reply:
I updated `getBulkItems` to surround the slice `await` with a try..catch block, so we should be returning + populating all the response data we get.

There's nothing built in to `_getBulkItems` to perform retries, but the external `bulkFetch` method can be called at a later point (the next refresh cycle, when we detect new items, whatever), and it has the smarts in it to only fetch ratings data for items we don't have ratings for.